igvm/drbd.py

```diff
@@ -267,44 +271,57 @@ def replicate_from_master(self, transaction=None):
             raise
 
     def wait_for_sync(self):
-        # Display a "nice" progress bar
-        show_progress = True
-        retry = 5
-        while show_progress:
-            lines = iter(self.hv.read_file('/proc/drbd').splitlines())
-            for line in lines:
-                line = line.decode()
-                if '{}: cs:'.format(self.get_device_minor()) in line:
-                    if 'ds:UpToDate/UpToDate' in line:
-                        show_progress = False
-                    try:
-                        next(lines)
-                        line = next(lines).decode().lstrip()
-                    except StopIteration:
-                        if retry:
-                            log.info(
-                                'No status yet, trying {} more time'
-                                .format(retry)
-                            )
-                            retry -= 1
-                            break
-                        log.warning(
-                            'Could not find progress bar, '
-                            'migrating without it!'
-                        )
-                        show_progress = False
-                    else:
-                        log.info(line)
-                    break
-            else:
-                # Exit the loop if status for current device can't be found
-                show_progress = False
-            sleep(1)
+        rep, size, _ = self._get_statistics()
+        last_n_synced = 0
+
+        with tqdm(
+            desc=self.vm_name,
+            total=size,
+            unit='iB',
+            unit_scale=True,
+            unit_divisor=1024,
+        ) as pbar:
+            while rep != 'Established' and rep is not None:
```

Comment:
The original code was handling a case where progress bar was not yet available because the receiving side has not yet started syncing. Is this handled with the new code too?

Reply:
we don't need to handle it anymore. as soon as you run "drbdadm up" the command will have statistics available